src/site/content/en/blog/promises/index.md

```diff
@@ -0,0 +1,1210 @@
+  an operation that hasn't completed yet.
+date: 2013-12-16
+tags:
+  - javascript
+authors:
+  - jakearchibald
+---
+
+Developers, prepare yourself for a pivotal moment in the history of
+web development.
+
+<em>[Drumroll begins]</em>
+
+Promises have arrived natively in JavaScript!
+
+<em>[Fireworks explode, glittery paper rains from above, the crowd goes wild]</em>
+
+At this point you fall into one of these categories:
+
+* People are cheering around you, but you're not sure what all the fuss is
+  about. Maybe you're not even sure what a "promise" is. You'd shrug, but the
+  weight of glittery paper is weighing down on your shoulders. If so, don't
+  worry about it, it took me ages to work out why I should care about this
+  stuff. You probably want to begin at the [beginning](#whats-all-the-fuss-about).
+* You punch the air! About time right? You've used these Promise things before
+  but it bothers you that all implementations have a slightly different API.
+  What's the API for the official JavaScript version? You probably want to begin
+  with the [terminology](#promise-terminology).
+* You knew about this already and you scoff at those who are jumping up and
+  down like it's news to them. Take a moment to bask in your own superiority,
+  then head straight to the [API reference](#promise-api-reference).
+
+## What's all the fuss about? {: #whats-all-the-fuss-about }
+
+JavaScript is single threaded, meaning that two bits of script cannot run at
+the same time; they have to run one after another. In browsers, JavaScript
+shares a thread with a load of other stuff that differs from browser to
+browser. But typically JavaScript is in the same queue as painting, updating
+styles, and handling user actions (such as highlighting text and interacting
+with form controls). Activity in one of these things delays the others.
+
+As a human being, you're multithreaded. You can type with multiple fingers,
+you can drive and hold a conversation at the same time. The only blocking
+function we have to deal with is sneezing, where all current activity must
+be suspended for the duration of the sneeze. That's pretty annoying,
+especially when you're driving and trying to hold a conversation. You don't
+want to write code that's sneezy.
+
+You've probably used events and callbacks to get around this. Here are events:
+
+```js
+var img1 = document.querySelector('.img-1');
+
+img1.addEventListener('load', function() {
+  // woo yey image loaded
+});
+
+img1.addEventListener('error', function() {
+  // argh everything's broken
+});
+```
+
+This isn't sneezy at all. We get the image, add a couple of listeners, then
+JavaScript can stop executing until one of those listeners is called.
+
+Unfortunately, in the example above, it's possible that the events happened
+before we started listening for them, so we need to work around that using
+the "complete" property of images:
+
+```js
+var img1 = document.querySelector('.img-1');
```

Comment:
Since `var` is assotiated with hoisting, variable redeclaration and scoping issues I'd suggest replacing all the "`vars`" in this file for `let`
```suggestion
let img1 = document.querySelector('.img-1');
```

Reply:
@IvanBila agreed. Would you like to do it in a subsequent PR?